lib/vm/src/vmcontext.rs

```diff
@@ -22,7 +22,7 @@ pub struct VMFunctionImport {
     /// A pointer to the imported function body.
     pub body: *const VMFunctionBody,
 
-    /// A pointer to the `VMContext` that owns the function.
+    /// A pointer to the `VMContext` that owns the function or host data.
```

Comment:
Can't we use an `enum` to differentiate between the two? In the pre-1.0 era, we used `vmctx` as an opaque pointer to hold various kinds of values, and it was really complex to debug and clearly error-prone. I don't want to reproduce this here.

Reply:
To clarify, this PR does nothing new in regards to vmctx, it simply documents what we've been doing from the start.

I agree that I'd also like to avoid it.  #1753 at least makes the behavior explicit (it's really bad that it's not even obvious what it is....).

I don't really see a reasonable alternative given our code right now, this is low level code that our generated code interacts with.  But this double use of a pointer is a high risk thing to do from a security perspective and I don't feel like it's simple enough to audit our code even with #1753.  We'd have to change the way generated code works I think